### ISSUE TITLE
Make dynamics training end-to-end

### ISSUE THREAD

**Issue by aryavohra** [PAL-ML/PAL_acme_ray#6]
**Currently, our dynamics model is trained via a reconstruction loss.** As we move to end-to-end training of the dynamics model, we will need to change our loss function. This section looks short but is likely to take a considerable amount of work.

![image](https://user-images.githubusercontent.com/8716483/131238788-b304fcc4-62cb-4c08-aaf9-c786dc0920b6.png)

- [ ] Make our trainer perform a K-step unroll as shown in MuZero's Figure 1. We need to do this to compute the predictions for policy, value, and reward.

![image](https://user-images.githubusercontent.com/8716483/131238813-a094b1c6-f8fd-4edf-b245-76ec27ca70af.png)
- [ ] Make our loss function use the predictions from the unroll to calculate an overall loss. See Equation 1 in Section 3 for the overall loss; the preceding explanation walks through it. Note that they use a cross-entropy loss for the rewards, values, and policy.
